commons/pkg/grpcclient/token_test.go

```
@@ -0,0 +1,158 @@
+
+	return tokenPath
+}
+
+func TestTokenInterceptor_AttachesToken(t *testing.T) {
+	tokenPath := writeTestToken(t, "test-token-value")
+	interceptor := TokenInterceptor(tokenPath)
+
+	var capturedCtx context.Context
+
+	fakeInvoker := func(ctx context.Context, _ string, _, _ any, _ *grpc.ClientConn, _ ...grpc.CallOption) error {
+		capturedCtx = ctx
```

golangci-lint (2.12.2)

[error] nested context in function literal

(fatcontext)